src/Storage/s3store.jl

```diff
@@ -76,9 +86,38 @@ end
 
 function cloud_list_objects(s::S3Store)
   prefix = (isempty(s.store) || endswith(s.store,"/")) ? s.store : string(s.store,"/")
-  listfun = s.listversion==2 ? S3.list_objects_v2 : S3.list_objects
-  listfun(s.aws, Bucket=s.bucket, prefix=prefix, delimiter = "/")
+
+  if s.backend == awss3
+    # TODO: This doesn't list subdirectories:
+    # objects = collect(s3_list_objects(s.aws, s.bucket, prefix))
+    #
+    # https://github.com/JuliaCloud/AWSS3.jl/pull/85
+    # Dict(
+    #   "Contents" => filter(e -> haskey("Key"), objects),
+    #   "CommonPrefixes" => filter(e -> haskey("Prefix"), objects),
+    # )
+
```

Comment:
A problem with `AWSS3` is that it doesn't return `CommonPrefixes`. I've opened an upstream PR: https://github.com/JuliaCloud/AWSS3.jl/pull/85